src/components/higlass/HiGlassSubscriber.js

```diff
@@ -111,16 +123,98 @@ export default function HiGlassSubscriber(props) {
   // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [loaders, dataset]);
 
-  const hgComponent = useMemo(() => (
-    <HiGlassComponent
-      zoomFixed={false}
-      viewConfig={hgViewConfig}
-      options={{
-        ...hgOptions,
-        theme,
-      }}
-    />
-  ), [hgViewConfig, hgOptions, theme]);
+  const hgOptions = useMemo(() => ({
+    ...hgOptionsProp,
+    theme,
+  }), [hgOptionsProp, theme]);
+
+  const hgViewConfig = useMemo(() => {
+    // HiGlass needs the start and end absolute genome coordinates
+    const initialXDomain = [
+      genomicTargetX - ((genomeSize / (2 ** genomicZoomX))) * ((width / 2) / HG_SIZE),
+      genomicTargetX + ((genomeSize / (2 ** genomicZoomX))) * ((width / 2) / HG_SIZE),
+    ];
+    const initialYDomain = [
+      genomicTargetY - ((genomeSize / (2 ** genomicZoomY))) * ((height / 2) / HG_SIZE),
+      genomicTargetY + ((genomeSize / (2 ** genomicZoomY))) * ((height / 2) / HG_SIZE),
+    ];
+    return {
+      editable: false,
+      zoomFixed: false,
+      trackSourceServers: [
+        '//higlass.io/api/v1',
+      ],
+      exportViewUrl: '//higlass.io/api/v1/viewconfs',
+      views: [
+        {
+          ...hgViewConfigProp,
+          initialXDomain,
+          initialYDomain,
+        },
+      ],
+      zoomLocks: {
+        locksByViewUid: {},
+        locksDict: {},
+      },
+      locationLocks: {
+        locksByViewUid: {},
+        locksDict: {},
+      },
+      valueScaleLocks: {
+        locksByViewUid: {},
+        locksDict: {},
```

Comment:
Is this for multiple instances but if you are using HiGlass for coordination?  If so, is there a way to hook our coordination model into this somehow?  Would that help performance?  I think you mentioned something about this in the Vitessce meeting but I just want it written in Github so it can be searchable for later.

Reply:
Yes it would help with performance if we could hook into the HiGlass internal view coordination at a lower level than the HiGlass view config. I tried hooking into the HiGlass `location` callback (which is not debounced) but it did not always return a result consistent with the values in the view config. I think I will try that again though, I may have been doing something wrong